backend/app/model/mixins/representative_file_version.rb

```diff
@@ -49,17 +65,44 @@ def sequel_to_jsonmodel(objs, opts = {})
           end
         end
 
-        # this could get very ugly
+        if json["representative_file_version"].nil? && self.name == "DigitalObject"
```

Comment:
Minor: maybe a comment here to explain something like ~ "If we are a digital object and have not found a representative file version yet lets check our components" -- helps me at least understand what is happening :smile:

Reply:
Probably ignore this and the next comment, the situation is explained "up top":

```
# Finally, a DigitalObject can fall back to a descendent DigitalObjectComponent's representative file version
# if necessary, and a Resource can fall back to a descendent ArchivalObject's representative file version.
```

Reply:
🚬